components/resource_control/src/service.rs

```diff
@@ -35,65 +35,78 @@ const RETRY_INTERVAL: Duration = Duration::from_secs(1); // to consistent with p
+            self.revision = revision;
+            // remove all potential deleted groups with retain.
+            let vaild_groups: Vec<_> = groups.iter().map(|g| &g.name).collect();
+            self.manager.retain(|name, _g| vaild_groups.contains(&name));
+            drop(vaild_groups);
+            groups
+                .into_iter()
+                .for_each(|rg| self.manager.add_resource_group(rg));
+            // Secondly, start watcher at loading revision.
+            loop {
+                match self
+                    .pd_client
+                    .watch_global_config(RESOURCE_CONTROL_CONFIG_PATH.to_string(), self.revision)
+                {
+                    Ok(mut stream) => {
+                        while let Some(grpc_response) = stream.next().await {
+                            match grpc_response {
+                                Ok(r) => {
+                                    self.revision = r.get_revision();
+                                    r.get_changes()
+                                        .iter()
+                                        .for_each(|item| match item.get_kind() {
+                                            EventType::Put => {
+                                                match protobuf::parse_from_bytes::<ResourceGroup>(
                                                     item.get_payload(),
-                                                )
-                                            {
-                                                self.manager.add_resource_group(group);
+                                                ) {
+                                                    Ok(group) => {
+                                                        self.manager.add_resource_group(group);
+                                                    }
+                                                    Err(e) => {
+                                                        error!("parse put resource group event failed, name: {}, err: {:?}", item.get_name(), e);
+                                                    }
+                                                }
                                             }
-                                        }
-                                        EventType::Delete => {
-                                            self.manager.remove_resource_group(item.get_name());
-                                        }
-                                    });
-                            }
-                            Err(err) => {
-                                error!("failed to get stream"; "err" => ?err);
-                                let _ = GLOBAL_TIMER_HANDLE
-                                    .delay(std::time::Instant::now() + RETRY_INTERVAL)
-                                    .compat()
-                                    .await;
+                                            EventType::Delete => {
```

Comment:
it's no need to parse `delete event`, just remove `item.name`
A `delete event` from etcd only contains the deleted key and its modification revision.
And I impl this in `mocker/etcd.rs` func delete, so u can not pass ut.

Reply:
item.name is the etcd key which is the full etcd path, I choose to parse the data to avoid depending on the data path pattern.

Reply:
@HuSharp Do you think the current item name is the full path is a PD bug or feature

Reply:
> item.name is the etcd key which is the full etcd path

Oh, you're right. I forgot that...... this is because we use the **get prefix** to get all configs for "watch" and "load".
I think it makes sense to return only the name.

Reply:
> item.name is the etcd key which is the full etcd path

Oh, you're right. This is because we use the get prefix to get all configs for "watch" and "load".
I think it makes sense to return only the name, which parsing in pd to delete the prefix path

Reply:
> which parsing in pd to delete the prefix path.

The prefix path maybe not be suitable delete in the server side, because the grpc method called `watchglobalconfig` rather than only for the resource group.

Reply:
But because we support watch multiple name in a single request, if the names are separated with each other, the caller may still need to depend on the returned name to distinct between each other. Though I don't like to depend on the internal data key path in pd.

Reply:
But a delete event from etcd only contains the deleted key and its modification revision, not contains value
So you need to trim item.name with config path

Reply:
So, Is decode from data cannot work?

Reply:
Yes, ut can pass, because 
- adding the `reload` function before watch makes watch delete not work
- watch delete parses the null value and gets a "" instead of an error
@glorv

Reply:
Oh, I misinterpret the ectd doc. I see the etcd watch event contains a PrevKv field, so shall we return the PrevKv.Payload instead?

Reply:
Ok, after test, I found user PrevKv.Payload also doesn't work. So we can only depend on the key here.

Reply:
modify to use WithPrevKV() to get the previous key-value pair when get Delete Event, so there can parse data.
https://github.com/tikv/pd/pull/5932 @glorv @nolouch PTAL!